common/src/path.rs

```diff
@@ -259,6 +260,104 @@ pub fn shorten_path(path: &str, length: usize) -> Cow<'_, str> {
     Cow::Owned(rv)
 }
 
+/// Extensions to `Path` for handling `dSYM` directories.
+pub trait DSymPathExt {
+    /// Returns `true` if this path points to an existing directory with a `.dSYM` extension.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::path::Path;
+    /// use symbolic_common::DSymPathExt;
+    ///
+    /// assert!(Path::new("Foo.dSYM").is_dsym_dir());
+    /// assert!(!Path::new("Foo").is_dsym_dir());
+    /// ```
+    fn is_dsym_dir(&self) -> bool;
+
+    /// Resolves the path of the debug file in a `dSYM` directory structure.
+    ///
+    /// Returns `Some(path)` if this path is a dSYM directory according to [`is_dsym_dir`], and a
+    /// file of the same name is located at `Contents/Resources/DWARF/`.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
```

Comment:
oh, and I thought these were actually real tests. Would you mind changing them in a way that they use any of the fixtures that we have? Would be nice to have coverage here.

Reply:
These are doc tests, so I cannot use our fixtures (and it also makes little sense). These docs are modelled after the `std::path::Path` docs.

However, I agree that this should be tested, let me add some tests.